cpp/ql/src/Likely Bugs/Memory Management/UninitializedLocal.ql

```diff
@@ -72,6 +72,10 @@ VariableAccess commonException() {
   or
   result.getParent() instanceof BuiltInOperation
   or
+  // If the uninitialized use is inside an expression statement that has
+  // no side effect, ignore it
+  result.getParent*().(ExprStmt).isPure()
```

Comment:
This doesn't look correct to me, as this would also exclude expressions like `x + y`, which is probably not what we want. This should probably check for a `VariableAccess` with a void cast that is also an expression statement. This will also need some tests.